src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Windows.cs

```diff
@@ -178,20 +172,25 @@ private static int GetWindowsProductType()
         [DllImport("kernel32.dll", ExactSpelling = true)]
         private static extern int GetCurrentApplicationUserModelId(ref uint applicationUserModelIdLength, byte[] applicationUserModelId);
 
-        internal static uint GetWindowsVersion()
+        private static volatile Version s_windowsVersionObject;
```

Comment:
Volatile to avoid torn write?

Reply:
Yeah, want the object ref to be published after all its fields are set.

Reply:
Why don't we just use `Lazy` for it?